lib/rubocop/cop/rspec/capybara/visibility_matcher.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for boolean visibility in capybara finders.
+        #
+        # Capybara lets you find elements that match a certain visibiity using
+        # the `:visible` option. `:visible` accepts both boolean and symbols as
+        # values, however using booleans can have unwanted effects. `visible:
+        # false` does not find just invisible elements, but both visible and
+        # invisible elements. For expressiveness and clarity, use one of the
+        # symbol values, `:all`, `:hidden` or `:visible`.
+        # (https://www.rubydoc.info/gems/capybara/Capybara%2FNode%2FFinders:all)
+        #
+        # @example
+        #
+        #   # bad
+        #   expect(page).to have_selector('.foo', visible: false)
+        #
+        #   # bad
+        #   expect(page).to have_selector('.foo', visible: true)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :all)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :hidden)
+        #
+        #   # good
+        #   expect(page).to have_selector('.foo', visible: :visible)
+        #
+        class VisibilityMatcher < Cop
+          MSG = 'Use `:all`, `:hidden`, `:visible` instead of `true`, `false`.'
```

Comment:
Ideally I'd want `visible: true` to have a message "Use `:visible` instead of `true`" and the `visible: false` to have a message `Use `:all` or `:hidden` instead of `false`", but I don't know how to.

Reply:
It can be done by using two separate matchers and messages, you can check https://github.com/rubocop-hq/rubocop-rspec/blob/master/lib/rubocop/cop/rspec/message_spies.rb#L30 for an example.